### PR TITLE
Using shared CloseHandle interop

### PR DESCRIPTION
**Pull request by Maxwe11**
Some projects was using its own definition of `CloseHandle` interops. This commit changes it to use the shared definition from Common folder.

### CODE REVIEW

File: src/Common/src/Interop/Windows/mincore/Interop.DuplicateHandle.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+using System.Security;
+
+internal partial class Interop
+{
+    internal partial class mincore
+    {
+        [DllImport(Libraries.Handle, SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        [SecurityCritical]
+        internal static extern bool DuplicateHandle<TProcessHandle, TSourceHandle, TTargetHandle>(TProcessHandle hSourceProcessHandle,
```
Comment (stephentoub) [inline src/Common/src/Interop/Windows/mincore/Interop.DuplicateHandle.cs]: @jkotas, is there any issue using generics in a P/Invoke like this?

File: src/Common/src/Interop/Windows/mincore/Interop.DuplicateHandle.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+using System.Runtime.InteropServices;
+using System.Security;
+
+internal partial class Interop
+{
+    internal partial class mincore
+    {
+        [DllImport(Libraries.Handle, SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
```
Comment (stephentoub) [inline src/Common/src/Interop/Windows/mincore/Interop.DuplicateHandle.cs]: MarshalAs(UnmanagedType.Bool) is the default for bool (and matches CloseHandle returning a BOOL), so this shouldn't be necessary.

File: src/Common/src/Interop/Windows/mincore/Interop.CloseHandle.cs [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+using System;
+using System.Runtime.InteropServices;
+using System.Security;
+
+internal partial class Interop
+{
+    internal partial class mincore
+    {
+        [DllImport(Libraries.Handle, SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
```
Comment (stephentoub) [inline src/Common/src/Interop/Windows/mincore/Interop.CloseHandle.cs]: The return attribute shouldn't be necessary here.

Comment (stephentoub) [inline src/Common/src/Interop/Windows/mincore/Interop.CloseHandle.cs]: Though I see that there is (or at least used to be) an FxCop rule that warns if you don't use MarshalAs on all Booleans in P/Invokes.  If that rule is still around, I guess we should leave this in.

**Comment by stephentoub** [comment]
As long as Jan's ok with the generics in the P/Invoke, LGTM.

**Comment by stephentoub** [comment]
ps Eventually we'll be moving all of these P/Invokes to Common, whether or not they're currently shared (more of them will be shared as more projects are released in the repo).  Thanks for cleaning up the currently shared ones.

File: src/Common/src/Interop/Windows/mincore/Interop.DuplicateHandle.cs (hunk 1 above)
Comment (jkotas) [inline src/Common/src/Interop/Windows/mincore/Interop.DuplicateHandle.cs]: Generic methods are not supported for P/Invoke. If you run this code, you are going to get:

System.TypeLoadException: Generic method or method in generic class is internal call, PInvoke, or is defined in a COM Import class.

Comment (stephentoub) [inline src/Common/src/Interop/Windows/mincore/Interop.DuplicateHandle.cs]: Which is further concerning, since it appears CI finished successfully.

Comment (stephentoub) [inline src/Common/src/Interop/Windows/mincore/Interop.DuplicateHandle.cs]: Turns out it the CI build actually failed, but it reported success.  Will follow-up with a separate issue.

**Comment by Maxwe11** [comment]
removed erroneous interop

**Comment by stephentoub** [comment]
LGTM

**Comment by stephentoub** [comment]
@pallavit (or @ellismg), could you peek at this?

**Comment by ellismg** [comment]
LGTM as well.  Thanks for cleaning this up!

**Pull request merged**